tests/test_worker.ts

```diff
@@ -710,6 +709,41 @@ describe('workers', function () {
     await worker.close();
   });
 
+
+  describe('when prioritized job is added while processing last active job', () => {
+    it('should process prioritized job whithout delay', async function () {
+      this.timeout(1000);
+      await queue.add('test1', { p: 2 }, { priority: 2 });
+      let counter = 0;
+      let processor;
+      const processing = new Promise<void>((resolve, reject) => {
+        processor = async (job: Job) => {
+          try {
+            if (job.name == 'test1')
+              {await queue.add('test', { p: 2 }, { priority: 2 });}
+
+            expect(job.id).to.be.ok;
+            expect(job.data.p).to.be.eql(2);
+          } catch (err) {
+            reject(err);
+          }
+
+          if (++counter === 2) {
+            resolve();
+          }
+        };
+      });
+
+      const worker = new Worker(queueName, processor, { connection });
```

Comment:
Can't it be that you're accessing `processor` before it is assigned?

Reply:
same comment as above